src/pipeline_v3/utils/security.py

```
@@ -0,0 +1,462 @@
+
+        # Remove/escape potentially dangerous characters for shell commands
+        # Allow common punctuation but escape shell metacharacters
+        dangerous_chars = ["$", "`", "\\", "\n", "\r", ";", "|", "&", ">", "<", "(", ")", "{", "}"]
+        for char in dangerous_chars:
+            value = value.replace(char, f"\\{char}")
+
+        # Limit length to prevent DoS
+        max_length = 1000
+        if len(value) > max_length:
+            value = value[:max_length]
+            logger.warning(f"Metadata value truncated to {max_length} characters")
```

Pylint (3.3.7)

[warning] Use lazy % formatting in logging functions

(W1203)